src/_pytest/python.py

```diff
@@ -1702,3 +1702,22 @@ def runtest(self) -> None:
         raise RuntimeError("function definitions are not supposed to be run as tests")
 
     setup = runtest
+
+
+@hookimpl(tryfirst=True)
```

Comment:
I don't think it's obvious why this is necessary, so I'll try to make a quick explanation:

Since the `assertion` plugin is always installed by default, and it is installed before this one, he'll always get preferred. It's implementation has [this code](https://github.com/pytest-dev/pytest/blob/bfbf733a3ed2f58a71570aa6e950444f3c5b0ea1/src/_pytest/assertion/util.py#L192-L193), which changes the message for "high" verbosity level, regardless of the lhs and rhs types, so I need this `tryfirst` here so that I can handle the message even with verbosity level > 0...

Reply:
Thanks for the explanation (this would be a good addition to the docstring of the function btw).

This makes me wonder if we should just bite the bullet and add this logic to `assertion.pytest_assertrepr_compare`.

@RonnyPfannschmidt @bluetech opinions?

Reply:
I added the docs. Will keep the conversation, thought, just to see if people want to discuss further.

Reply:
> This makes me wonder if we should just bite the bullet and add this logic to assertion.pytest_assertrepr_compare.

Given that `approx` is built into pytest, and is not in a plugin, I think it does make sense. Currently the hook is defined in the `python` plugin which is not really related.

However, if it's too much work it's OK to leave as-is for now, maybe add a TODO.